controllers/page-routes.js

```diff
@@ -1,22 +1,162 @@
+      where: {
+        user_id: req.session.user_id,
+        cancelled: false,
+      },
+    });
+    const services = serviceData.map((service) => service.get({ plain: true }));
+
+    const userData = await User.findAll({
+      include: {
+        model: Industry,
+      },
+      attributes: { exclude: ['password'] },
+      where: {
+        user_id: req.session.user_id,
+      },
+    });
+
+    const user = userData.get({ plain: true });
+
+    res.render('dashboard', {
+      bookings,
+      services,
+      user,
+      logged_in: req.session.logged_in,
+    });
+  } catch (err) {
+    res.render('error', {
+      status: 500,
+      message: 'A internal server error has occured',
+      logged_in: req.session.logged_in,
+    });
+  }
+});
+
+router.get('/signin', (req, res) => {
+  if (req.session.logged_in) {
+    res.redirect('dashboard');
+    return;
+  }
+  res.render('signin');
+});
+
+router.get('/register', (req, res) => {
+  if (req.session.logged_in) {
+    res.redirect('dashboard');
+    return;
+  }
+  res.render('register');
+});
+
+router.get('/search', async (req, res) => {
+  const serviceData = await Service.findAll({
+    
+    where: {
+      user_id: req.session.user_id,
+      cancelled: {[Op.not]: true}, 
+    },
+  });
+  const services = serviceData.map((service) => service.get({ plain: true }));
+
+  res.render('search', {
+    services,
+    logged_in: req.session.logged_in,
+  });
+});
+
+router.get('/profile', withAuth, async (req, res) => {
+  const profileData = await User.findByPk(req.session.user_id, {
+    attributes: { exclude: ['password'] },
+  });
+
+  const profile = profileData.get({ plain: true });
+
+  res.render('profile', {
+    profile,
+    logged_in: req.session.logged_in,
+  });
 });
 
-router.get("/signin", (req, res) => {
-    res.render("signin");
+router.get('/edit-profile', withAuth, async (req, res) => {
+  const editProfileData = await User.findByPk(req.session.user_id, {
+    attributes: { exclude: ['password'] },
+  });
+
+  const profile = editProfileData.get({ plain: true });
+
+  res.render('profile', { profile, logged_in: req.session.logged_in });
 });
 
-router.get("/register", (req, res) => {
-    res.render("register");
-})
+router.get('/service/:id', withAuth, async (req, res) => {
+  try {
+    const ServiceData = await Service.findByPk(req.params.id);
+    if (!ServiceData) {
+      res.render('error', {
+        status: 404,
+        message: 'A service with this ID cannot be found',
+        logged_in: req.session.logged_in,
+      });
+      return;
+    }
+    const service = ServiceData.get({ plain: true });
+    res.render('service', {
+      service,
+      logged_in: req.session.logged_in,
+    });
+  } catch (err) {
+    res.render('error', {
+      status: 500,
+      message: 'A internal server error has occured',
+      logged_in: req.session.logged_in,
+    });
+  }
+});
+
+router.get('/service/:id?mode=edit|create', withAuth, async (req, res) => {

```

Comment:
```suggestion
router.get('/edit-service', withAuth, async (req, res) => {

```
This route was specified differently in the document. Also, the query string parameter (? and everything after) is not included in the route definition, it is parsed in the function body by the res.query object. 

On reflection we need to change this route anyway as we are not going to know the ID of the service if we are creating a new one and thus was wrong when we specified it originally.  This page will have requests that look like the following: 

/edit-service?mode=create
/edit-service?mode=edit&id=5

We need to check if req.query contains mode or id. In the event of mode=create, render the edit-service page with just the logged_in status. If it contains mode=edit and no id value, call and internal server error as we don't know what we are editing and if it contains mode=edit and has an ID, look up the ID from the database and add it to the render call of the edit-service page.

We also need to check if the user attempting to edit a service is the same one who posted the service or we need to throw an error saying that they are unauthorised to edit the service. 

For reference of working with query strings: https://expressjs.com/en/api.html#req.query